check_sites.py

```diff
@@ -53,20 +68,38 @@ def main():
     error_texts = []
 
     try:
-        fps_checker.validate_schema(os.path.join(inputPrefix,'SCHEMA.json'))
+        fps_checker.validate_schema(os.path.join(input_prefix,'SCHEMA.json'))
     except Exception as inst:
         # If the schema is invalid, we will not run any other checks
         print(inst)
         exit()
 
     check_sets = {}
-    try:
+    # If called with with_diff, we must determine the sets that are different 
+    # to properly construct our check_sets
+    if with_diff:   
+        with open(os.path.join(input_prefix,'first_party_sets.JSON')) as f:
+            try:
+                old_sites = json.load(f)
+            except Exception as inst:
+            # If the file cannot be loaded, we will not run any other checks
+                print("There was an error when loading " +
+                    os.path.join(input_prefix,'first_party_sets.JSON') + 
+                    "\nerror was: " + inst)
+                exit()
+        old_checker = FpsCheck(old_sites, etlds, icanns)
+        check_sets, _ = find_diff_sets(old_checker.load_sets(), fps_checker.load_sets())
+        # TODO: add variable and check for subtracted_sets in case of user 
+        # removing old set from the list
+    else:
         check_sets = fps_checker.load_sets()
-    except Exception as inst:
-        error_texts.append(inst)
 
+    
+    # Run rationale check separately from check_list since it takes no argument
+    fps_checker.has_all_rationales
```

Comment:
Why is this step different? Shouldn't they all either accept the list of sets that they're validating, or all read from the same member variable of sets?